### DISCUSSION TITLE
Document how to install from a Git clone

### DISCUSSION THREAD

**Discussion by akaihola** [marienz/nix-doom-emacs-unstraightened#16] [Ideas]
I ran into an error (see #15) and would like to try and find a commit which introduced this problem.

For a home-manager install, the README file only shows how to use the Unstraightened home-manager module. It would be useful to also have instructions on how to install nix-doom-emacs-unstraightened from an arbitrary commit

**Comment by marienz** [comment]
You can pin an arbitrary commit by putting `nix-doom-emacs-unstraightened.url = "github:marienz/nix-doom-emacs-unstraightened/9205f2878ff2d173b91cf27f594d4f9a983b639f` in your flake.nix and running `nix flake lock --update-input nix-doom-emacs-unstraightened` (older Nix) / `nix flake update nix-doom-emacs-unstraightened` (newer Nix).

Alternatively, to use a specific commit just once, pass `--override-input nix-doom-emacs-unstraightened github:marienz/nix-doom-emacs-unstraightened/9205f2878ff2d173b91cf27f594d4f9a983b639f` to `nix` (or `nixos-rebuild`, which conveniently passes it through).

(Note that at first glance your failure in #15 occurs before Unstraightened is actually involved, so trying to bisect that failure may not work.)

**Comment by akaihola** [comment]
Thanks @marienz for your advice!

I suppose things should look a bit different as I have a stand-alone install of home-manager, not using flakes. I'll try to figure it out – as a Nix newbie I'll learn a lot on the way.

I'm continuing debugging my install in #15. Closing this discussion.
